Data_cleaning/load_clean_concat.ipynb

```diff
@@ -200,15 +295,36 @@
     "    for file_name in file_names:\n",
     "        df = pd.read_csv(data_folder+file_name+'.csv', parse_dates=['TIMESTAMP'])\n",
     "        df.rename(mapper=remove_pit_suffix, axis='columns', inplace=True)\n",
-    "            \n",
+    "        \n",
     "        if print_text_result: print(f'===== {file_name} =====')\n",
+    "        \n",
+    "        diff_dict = dict()\n",
+    "        prune_unnecessary_features(df)\n",
+    "        \n",
+    "        diff_dict['dates_removed'] = filter_dates(df, file_name)\n",
+    "        diff_dict['duplicates_removed'] = filter_duplicate_date(df)\n",
+    "        \n",
+    "        # here is where the wavelet coefficients are calculated \n",
+    "        df = fill_na_redox_values(df, file_name)\n",
+    "        df = calculate_wavelet_coefficients(df, 1/2, 5, file_name)\n",
+    "        \n",
+    "        window_sizes = get_time_windows([12, 24])\n",
+    "        \n",
+    "        td = timedelta(days = 0, hours = 0, minutes = 5, seconds = 0, milliseconds= 0, weeks = 0)\n",
+    "        \n",
+    "        if 'VII' not in file_name:\n",
+    "            df = sigma_feature_engineering(df, window_sizes=window_sizes, td=td)\n",
+    "        \n",
+    "        diff_dict['missing_data_removed'] = filter_missing_values(df, file_name)\n",
```

Comment:
Filtering missing values after sigma_feature_engineering can cause miss conception with the sigma values as values in between are removed and the sigma values are depending on previous measurement (5min time frame). Now some values in between might be removed and there can exist +5min difference in time.

I'm not sure if this is a issue but just adding an open question

Reply:
I disagree. Removing the rows with missing values before calculating sigma features will actually create more gaps due to other features missing (not redox values) - having more gaps means many more rows removed when calculating sigma features

Reply:
Good point! Lets keep it as it is  :+1: